rotkehlchen/utils/network.py

```
@@ -35,6 +35,9 @@ def create_session() -> requests.Session:
         connect=3,
         read=2,
         status=0,
+        # in the case of 500 errors or rate limits don't raise error and return the response.
+        # We handle manually error status codes.
+        raise_on_status=False,
```

1. Why are you making this change? To not raise on 429? Are you 100% sure this does not affect any other part of the code that uses `create_session? This sounds quite dangerous.
2. If you want to configure this properly then add all the required arguments along with an explanation in proper Engilish of what is it you are trying to achieve through all arguments.

The relevant arguments are

```

    :param Collection allowed_methods:
        Set of uppercased HTTP method verbs that we should retry on.

        By default, we only retry on methods which are considered to be
        idempotent (multiple requests with the same parameters end with the
        same state). See :attr:`Retry.DEFAULT_ALLOWED_METHODS`.

        Set to a ``None`` value to retry on any verb.

    :param Collection status_forcelist:
        A set of integer HTTP status codes that we should force a retry on.
        A retry is initiated if the request method is in ``allowed_methods``
        and the response status code is in ``status_forcelist``.

        By default, this is disabled with ``None``.


    :param bool raise_on_redirect: Whether, if the number of redirects is
        exhausted, to raise a MaxRetryError, or to return a response with a
        response code in the 3xx range.

    :param bool raise_on_status: Similar meaning to ``raise_on_redirect``:
        whether we should raise an exception, or return a response,
        if status falls in ``status_forcelist`` range and retries have
        been exhausted.

```

and don't rely on defaults. Defaults can change. For more arguments read the `Retry` docs.

---

1. The change is because as it is right now in the code 429 raises `RetryError` from requests instead of returning the response with the status code 429 and our custom logic to read the headers and retry is not used. I've checked and where we use this session we check the status code of the responses for errors so this change shouldn't break anything.